File: Unidad_3/Java/Programa21_U3.java
package com.mycompany.programa21_u3;

import java.util.Scanner;
public class Programa21_U3 {

public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingrese la cantidad de números: ");
        int cantidadNumeros = scanner.nextInt();

        int contador = 1;

        while (contador <= cantidadNumeros) {
            System.out.print("Ingrese un número natural positivo: ");
            int numero = scanner.nextInt();

            if (numero > 0) {
                int cubo = numero * numero * numero;
                System.out.println("El cubo de " + numero + " es " + cubo);
            } else {
                System.out.println("Por favor, ingrese un número natural positivo.");
            }

            contador++;
        }
    }
}

File: Unidad_3/C#/Programa21_U3.cs
using System;

namespace Programa21_U3
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.Write("Ingrese la cantidad de números: ");
            int cantidadNumeros = Convert.ToInt32(Console.ReadLine());

            int contador = 1;

            while (contador <= cantidadNumeros)
            {
                Console.Write("Ingrese un número natural positivo: ");
                int numero = Convert.ToInt32(Console.ReadLine());

                if (numero > 0)
                {
                    int cubo = numero * numero * numero;
                    Console.WriteLine("El cubo de " + numero + " es " + cubo);
                }
                else
                {
                    Console.WriteLine("Por favor, ingrese un número natural positivo.");
                }

                contador++;
            }
        }
    }
}